service/src/main/kotlin/app/cash/backfila/service/runner/BackfillRunner.kt

```diff
@@ -99,7 +100,12 @@ class BackfillRunner private constructor(
   fun run() {
     factory.loggingSetupProvider.withLogging(backfillName, backfillRunId, partitionName) {
       runBlocking(MDCContext()) {
-        start(this)
+        try {
+          start(this)
+        } finally {
+          logger.info { "Runner finalizing backfill: ${logLabel()}" }
+          finalize()
```

Comment:
this isn't the end of the backfill. this can happen when the pod shuts down, or the backfill is paused. we need to do this in the code that sets it to COMPLETE state

Reply:
We need a couple tests around this too.